src/App.tsx

```diff
@@ -1,60 +1,106 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable react/jsx-no-comment-textnodes */
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  function getNewPostId(todos: Todo[]) {
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    return maxId + 1;
+  }
+
+  const addNewTodo = (newTodo: Todo) => {
+    setTodos(todos => [...todos, newTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (evnt: React.FormEvent) => {
+    evnt.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(userId === 0);
+
+    if (!title.trim() || !userId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNewPostId(todos),
+      title,
+      userId,
+      completed: false,
+    };
+
+    addNewTodo(newTodo);
+
+    reset();
+  };
```

Comment:
[CODE STYLE] - Don't repeat yourself. The error handling for 'title' and 'userId' is being set in both 'handleSubmit' and 'handleTitleChange'/'handleUserChange'. Consider creating a separate function to handle the validation logic to avoid repetition.